src/assets/scripts/client/aircraft/AircraftCommander.js

```diff
@@ -369,16 +376,67 @@ export default class AircraftCommander {
     }
 
     /**
+     * Used only for suffix routes.
+     *
+     * Suffix routes apply to a specific runway.
+     * This method will find and pass on the correct `RunwayModel`
+     * to the `Pilot`.
+     *
+     * @for AircraftCommander
+     * @method _runSIDforSuffix
+     * @param  aircraft {AircraftModel}
+     * @param airportModel {AirportModel}
+     * @param sidId {strig}
+     * @return {array}  [success of operation, readback]
+     */
+    _runSIDforSuffix(aircraft, airportModel, sidId) {
+        const routeModel = this._navigationLibrary.sidCollection.findRouteByIcao(sidId);
+        const runwayName = routeModel.getSuffixSegmentName(PROCEDURE_TYPE.SID);
+        const runwayModel = airportModel.getRunway(runwayName);
+
+        return aircraft.pilot.applyDepartureProcedure(sidId, runwayModel, airportModel.icao);
+    }
+
+    /**
      * @for AircraftCommander
      * @method runSTAR
      * @param data {array<string>} a string representation of the STAR, ex: `QUINN.BDEGA2.KSFO`
+     * @return {array}   [success of operation, readback]
      */
     runSTAR(aircraft, data) {
         const routeString = data[0];
-        const arrivalRunway = aircraft.fms.arrivalRunwayModel;
-        const { name: airportName } = this._airportController.airport_get();
+        // TODO: why are we passing this if we already have it?
+        const runwayModel = aircraft.fms.arrivalRunway;
+        const airportModel = this._airportController.airport_get();
+
+        if (this._navigationLibrary.isSuffixRoute(routeString, PROCEDURE_TYPE.STAR)) {
+            return this._runSTARforSuffix(aircraft, airportModel, routeString);
+        }
+
+        return aircraft.pilot.applyArrivalProcedure(routeString, runwayModel, airportModel.name);
```

Comment:
So if the string _includes_ a suffix, the appropriate procedure will be flown. If a suffix is _not specified_, the procedure will be run for the runway designated as active.

EXACTLY what I was looking for. And it worked perfectly in my trials. :+1: